### ISSUE TITLE
Detect incompatible Python environments missing sqlite3

### ISSUE THREAD

**Issue by petetronic** [posit-dev/positron#4698]
If you use pyenv to install Python 3 on Linux environments, including Fedora 40, and sqlite3 was not already installed on the OS, you may miss a warning during installation of the Python version that says something like this:
```
WARNING: The Python sqlite3 extension was not compiled. Missing the SQLite3 lib?
```

When you then go to install a Python package that requires this module, such as ipykernel, as do Python language runtimes being initialized for the first time in Positron, they will fail at runtime with an error:

```
Python 3.11.10 (Pyenv) exited unexpectedly with error: ModuleNotFoundError: No module named '_sqlite3'
```

If you install sqlite3 in the Linux operating system, and use pyenv to re-install that Python version, it will then not show the warning during installation and packages like ipykernel (and Positron Python language runtimes) will be successful at runtime.

Positron Python could detect this missing dependency and suggest to the user why the runtime could not start up.

**Comment by petetronic** [comment]
Testing something a command like `python -c "from _sqlite3 import *"` does not error would be a way to detect this issue.

**Comment by petetronic** [comment]
Confirmed fixed in Positron 2024.11.0 build 140 on Fedora 40 per the steps in https://github.com/posit-dev/positron/pull/5149#issuecomment-2438353417
